VoodooI2CHID/VoodooI2CHIDDevice.cpp

```diff
@@ -347,12 +347,17 @@ IOReturn VoodooI2CHIDDevice::setReport(IOMemoryDescriptor* report, IOHIDReportTy
     UInt16 size;
     UInt16 arguments_length;
     UInt8 report_id = options & 0xFF;
-    UInt8* buffer = (UInt8*)IOMalloc(report->getLength());
-    report->readBytes(0, buffer, report->getLength());
-
+    UInt16 report_length = report->getLength();
```

Comment:
The report_length may not always be 16 bytes in length.

See: https://developer.apple.com/documentation/kernel/iomemorydescriptor/1442036-getlength


We may have overflow issues here,

Reply:
We'll tackle this in a follow up PR.